### PR TITLE
v4 typedefs and migration guide

### PR DESCRIPTION
**Pull request by serges147**
- eliminated struct/enum typedef-s
- added migration guide

### CODE REVIEW

File: libcanard/canard.c [hunk 1]
```diff
@@ -1104,38 +1107,35 @@ int32_t canardTxPush(CanardTxQueue* const                que,
...
-CanardTxQueueItem* canardTxPop(CanardTxQueue* const que, const CanardTxQueueItem* const item)
+struct CanardTxQueueItem* canardTxPop(struct CanardTxQueue* const que, struct CanardTxQueueItem* const item)
 {
-    CanardTxQueueItem* out = NULL;
     if ((que != NULL) && (item != NULL))
     {
-        // Intentional violation of MISRA: casting away const qualifier. This is considered safe because the API
-        // contract dictates that the pointer shall point to a mutable entity in RAM previously allocated by the
-        // memory manager. It is difficult to avoid this cast in this context.
-        out = (CanardTxQueueItem*) item;  // NOSONAR casting away const qualifier.
```
Comment (serges147) [inline libcanard/canard.c:1127]: As far as we now have "mutable" item from `canardTxPeek`, I decided also change signature of `canardTxPop` by removing `const` from the `item` parameter. Hence no need anymore to violate MISRA. Gonna do similar change at Udpard as well.

File: MIGRATION_v3.x_to_v4.0.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+7. **Test Thoroughly**:
+   - After making the changes, thoroughly test your application to ensure that it functions correctly with the new library version.
+   - Pay special attention to memory management and potential leaks.
+
+## Conclusion
+
+Updating from Libcanard v3.x to v4.0 involves several significant changes, especially in memory management and API function prototypes. By following this guide and carefully updating your code, you can migrate your application to use the new version of the library successfully.
+
+For more detailed information, refer to the updated Libcanard documentation and consider reaching out to the maintainers or community if you encounter any issues.
```
Comment (pavel-kirienko) [inline MIGRATION_v3.x_to_v4.0.md:273]:
```suggestion
```

I think this is not needed

**Review by pavel-kirienko — APPROVED** [review]
Very decent. Was it o1 or 4o?

**Comment by serges147** [comment]
> Very decent. Was it o1 or 4o?

o1

File: MIGRATION_v3.x_to_v4.0.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    - [New Functions](#new-functions)
+    - [Modified Functions](#modified-functions)
+    - [Removed Functions](#removed-functions)
+  - [Data Structure Changes](#data-structure-changes)
+    - [Type Definitions](#type-definitions)
+    - [Struct Modifications](#struct-modifications)
+  - [Memory Management Changes](#memory-management-changes)
+  - [Migration Steps](#migration-steps)
+  - [Conclusion](#conclusion)
+
```
[the remark this answers is not in the document]
Comment (serges147) [reply MIGRATION_v3.x_to_v4.0.md:21]: ok, I will eliminate it

**Pull request merged**